### PR TITLE
devtools: backport modern security and symbol check tooling

### PR DESCRIPTION
**Pull request by patricklodder**
This backports the Bitcoin Core 24.0.1 `security-check.py` and `symbol-check.py` scripts to enable consistent (and extended) checks on our binaries, now and in the future. This is important because: 

- we're about to switch build environments and we have no consistent tooling for both the current Ubuntu Bionic and the future Ubuntu Focal build hosts
- we're currently not checking any Windows binaries for backward compatibility, even though this is the most used binary distribution and we're not able to check ARM binaries with the current toolset.

See https://github.com/dogecoin/dogecoin/pull/3185#issuecomment-1345588713 for additional context - this PR aims to replace all changes to binary checking scripts in that PR.

## Approach taken

I have taken an approach where I took the scripts used for the last Bitcoin Core release and then reconfigured these because:
1. It's seen a lot of real life testing (this is used inside the Bitcoin Core release process just like it is in ours) so it is not bleeding edge
2. Forward cherry-picking would entail tons of self-overwriting work, 42 commits in total, so that makes no sense and just adds pressure to reviewers, especially since Bitcoin Core has switched low level binary parsing frameworks twice since what we have today

Of course there are a lot of features on the binaries that Bitcoin Core has implemented and we have not yet done, and I decided to keep the code but disable the checks that I could not fix with parameterization. This resulted in:

### New features supported

- macOS binary security checks ( no undefs, no lazy bindings, canary check )
- Windows PIE security check and dynamic libraries symbol check
- Linux ELF interpreter checks, and support for checking ARM binaries

### Checks we cannot support (fully) yet

- control flow checks for all binaries - can be enabled for Windows/Linux once we have gcc8 or higher, for macOS we may do this now by configuring boost different (because we use clang)
- macOS symbol checks - needs build system patches
- Windows ASLR checks - needs to be implemented for other binaries than dogecoind.exe / dogecoin-qt.exe 
- Windows minimum version checks - needs to be set when we link the binaries

## Notes to the parametrization and integration

- `lief`, the library used to parse cross-platform binaries, is not available in binary distribution for Ubuntu Bionic and takes about an hour to compile on the CI, so to not make the test time explode, I have uploaded a precompiled version to depends.dogecoincore.org. Once we have migrated away from Bionic, we can simply use the pypi supplied whl files for Focal.
- `aarch64` binaries have a different glibc minimum version than everything else (2.17), but this version has been consistent since the first release with 1.14.0 in 2019, meaning that there is no regression by allowing this.
- Additional win32 and win64 needed dynamic libraries have been reverse engineered from the 1.14.6 release binaries.
- The `Makefile.am` implementation of `check-symbols` and `check-security` didn't work well and needed to be changed to allow non-linux builds to use these, so I have amended these in a separate commit.

## Review & testing

- ~~This PR is missing 1 commit: to integrate the new scripts into the current gitian descriptors (because they require python3). However, this depends on #3144 to be merged, and since as of right now that PR breaks gitian builds for osx, this PR, and the additional commit, is unfortunately blocked. I am therefore tagging `do not merge`~~ - resolved.
- It makes no sense to review all commits together because there is a literal file overwrite summarizing 6 years of work in the first commit. As these are needed as one bundled PR to fix every build system at once, I cannot split them up. Instead I'd recommend reviewing each commit sequentially - I tried making it as clean as I could.
- Please test this against past 1.14 release binaries as well as known non-compliant binary output: e.g. by building Dogecoin Core on Ubuntu Jammy and then running the checks.  
- This will need a gitian check before merge

## Thanks

Much ❤️ to all the awesome Bitcoin Core maintainers for creating this great set of checks. And big respect @xanimo for helping me quickly solve issues these past few days.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (chromatic) [inline .github/workflows/ci.yml:234]: Not that I mind the two lines here, but I think recent versions of `pip` can install from a URL.

File: contrib/devtools/symbol-check.py
Comment (chromatic) [inline contrib/devtools/symbol-check.py:58]: Are `environ` and `_environ` no longer exported? Is this cleanup we should have done earlier?

File: contrib/devtools/symbol-check.py
Comment (chromatic) [inline contrib/devtools/symbol-check.py:148]: I'm not sure I understand why the extension is lowercase in one example but the entire filename is lowercase in another. There's another example later.

Do we need a consistency path, or a case normalization mapping somewhere? Or do we handle this on an exceptional basis as we see them?

**Comment by chromatic** [comment]
First commit: makes sense, no concerns, we'll get CI feedback rapidly so no great risk there.

Second commit: no concerns, one small comment we can ignore for now (or for good).

Third commit: I've skimmed subsequent commits. Are we dropping the binary paths set in environment variables because `lief` performs the same behavior? I think that's the case per my reading of the [lief GH README](https://github.com/lief-project/LIEF). I think I see these symbols moved to the code in the sixth commit, but I' reading commit-by-commit and want to confirm our intent.

Fourth commit: Inline comments. Nothing blocking, but I'd like to understand a few things better.

Fifth commit: makes sense. Should we link to issues in these comments?

Sixth commit: no concerns not already raised with third commit.

I'll try to get to a gitian build before the end of the weekend and possibly the Jammy build.

**Comment by chromatic** [comment]
Here's a snippet out output on Jammy in a Docker container:

```
src/qt/dogecoin-qt: export of symbol _ZTSN5boost6system6detail21system_error_categoryE not allowed!
src/qt/dogecoin-qt: export of symbol _ZTIN5boost16thread_exceptionE not allowed!
src/qt/dogecoin-qt: export of symbol __libc_single_threaded not allowed!
src/qt/dogecoin-qt: export of symbol _ZTSN5boost15program_options6detail27common_config_file_iteratorE not allowed!
src/qt/dogecoin-qt: libboost_filesystem.so.1.74.0 is not in ALLOWED_LIBRARIES!
src/qt/dogecoin-qt: libboost_program_options.so.1.74.0 is not in ALLOWED_LIBRARIES!
src/qt/dogecoin-qt: libboost_thread.so.1.74.0 is not in ALLOWED_LIBRARIES!
```

If this looks like what you're expecting, I'll make the full output available.

File: contrib/devtools/symbol-check.py
Comment (xanimo) [reply contrib/devtools/symbol-check.py:148]: It's weird yeah I noticed the same thing a while back and brought that up but my understanding is that we can follow up with optimizations like uniform casing on the next round of updates.

**Comment by xanimo** [comment]
> Here's a snippet out output on Jammy in a Docker container:
> 
> ```
> src/qt/dogecoin-qt: export of symbol _ZTSN5boost6system6detail21system_error_categoryE not allowed!
> src/qt/dogecoin-qt: export of symbol _ZTIN5boost16thread_exceptionE not allowed!
> src/qt/dogecoin-qt: export of symbol __libc_single_threaded not allowed!
> src/qt/dogecoin-qt: export of symbol _ZTSN5boost15program_options6detail27common_config_file_iteratorE not allowed!
> src/qt/dogecoin-qt: libboost_filesystem.so.1.74.0 is not in ALLOWED_LIBRARIES!
> src/qt/dogecoin-qt: libboost_program_options.so.1.74.0 is not in ALLOWED_LIBRARIES!
> src/qt/dogecoin-qt: libboost_thread.so.1.74.0 is not in ALLOWED_LIBRARIES!
> ```
> 
> If this looks like what you're expecting, I'll make the full output available.

Hmm, last comment didn't post re casing of windows binary ext 🤔 but yeah this output is to be expected since it's designed as a ground truth for `bionic`. It will definitely fail on `jammy`.

File: contrib/devtools/symbol-check.py
Comment (xanimo) [reply contrib/devtools/symbol-check.py:148]: Yeah I was thinking of drafting something up in a follow up PR. I would think adding a function to normalize the binary.name on the check would be the most flexible so long as the name matches but that's just my initial thought. Still unclear why there's a discrepancy in the first place.

**Comment by xanimo** [comment]
I think making an aggregate issue is the most sensible direction and that the comments are fine but it wouldn't hurt, really up to you imo. 

One thing I just noticed is that if we're no longer conditionally requiring `--enable-glibc-back-compat`, the following should be removed [here](https://github.com/patricklodder/dogecoin/blob/1.14.7-better-binary-checks/configure.ac#L164-L168) and the `AM_CONDITIONAL` [here](https://github.com/patricklodder/dogecoin/blob/1.14.7-better-binary-checks/configure.ac#L1083), either in this or in a follow up PR.

**Comment by xanimo** [comment]
Right yeah that makes sense. My mistake.

**Comment by xanimo** [comment]
Wow, so just noting this for the benefit of all, I just passed symbol-check with this branch on my `focal` build box with no code changes. Initially ran into missing lib error on `-lrt`. The only change I did, in addition to fixing my configured compilers (`gcc-7-multilib`, `g++-7-multilib`, `i686-w64-mingw32-gcc`, `i686-w64-mingw32-g++`), was adding `CFLAGS=-pthread` to:
```
./configure --prefix=`pwd`/depends/i686-w64-mingw32 --enable-reduce-exports --enable-glibc-back-compat LDFLAGS=-static-libstdc++
```
Hoping is not a false positive but was built without any changes to the code, no addl. interface overrides, etc. :D Will do some more tests on some of the other architectures I was having issues with and a run through security check but so far very promising results!

File: contrib/devtools/symbol-check.py
Comment (xanimo) [reply contrib/devtools/symbol-check.py]: I ended up moving to `i686-pc-linux-gnu` last night and got stuck on that per lack of interface overrides which is to be expected. Let me start a remote build with 64-bit mingw to confirm.

**Comment by xanimo** [comment]
Qt took forever for 64-bit mingw depends re the .dll/.DLL inquiry so putting on pause while I start a gitian build of this. Allocating all my resources to it so hopefully it be pretty quick.

**Comment by xanimo** [comment]
Hmm, I'm getting:
```
From https://github.com/patricklodder/dogecoin
 * branch            9d247ddfbb50f5630c8502cc77d76ae6f7b93113 -> FETCH_HEAD
--- Building for bionic amd64 ---
Stopping target if it is up
Making a new image copy
Starting target
Checking if target is up.
Preparing build environment
lstat /home/bluezr/source/repos/gitian-build-1-16-23/gitian-builder/inputs/lief-0.12.3-cp36-cp36m-linux_x86_64.whl: no such file or directory
Updating apt-get repository (log in var/install.log)
Installing additional packages (log in var/install.log)
Upgrading system, may take a while (log in var/install.log)
Creating package manifest
Creating build script (var/build-script)
Running build script (log in var/build.log)
Traceback (most recent call last):
        6: from ./bin/gbuild:347:in `<main>'
        5: from ./bin/gbuild:347:in `each'
        4: from ./bin/gbuild:349:in `block in <main>'
        3: from ./bin/gbuild:349:in `each'
        2: from ./bin/gbuild:354:in `block (2 levels) in <main>'
        1: from ./bin/gbuild:187:in `build_one_configuration'
./bin/gbuild:23:in `system!': failed to run on-target setarch x86_64 bash -x < var/build-script > var/build.log 2>&1 (RuntimeError)
```

**Comment by xanimo** [comment]
> > lstat /home/bluezr/source/repos/gitian-build-1-16-23/gitian-builder/inputs/lief-0.12.3-cp36-cp36m-linux_x86_64.whl : no such file or directory
> 
> Did you use the updated `gitian-build.sh` I provided in 9d247dd?

I did not. That did the trick! 🤦 Thank you kindly! 😅

**Comment by xanimo** [comment]
I just ran out of space on my gitian box at `aarch64-linux-gnu`. Have to clear some cache, clean up stuff but it looks as though it happened when moving from `../gitian-builder/build/out` to `gitian-output` so I think was just in the nick of time:

```
a202d4c8f3602704f276e17d088e1c58d6cea343ac7f76bf5abcb08f59c47751  dogecoin-1.14.7-osx-unsigned.dmg
e9dacbd2bb121dfa906e51fa15086ad48e052aaf8ae956a96f78e6b0f51413a3  dogecoin-1.14.7-osx-unsigned.tar.gz
14f8ad1074e27db6a334e1b250ebff03bab0d3bb9d9e5163ee32fa022755a26d  dogecoin-1.14.7-osx64.tar.gz
0318f6b9578fd9de9770bf8042f7397237a8ec89b34b6d2021e4301142379bc5  src/dogecoin-1.14.7.tar.gz
c838c0211a95d4c81ffb343779d24bb3a81536509b8a8dda0782d19f175844d9  dogecoin-1.14.7-win-unsigned.tar.gz
3eb24ccbd3b8b7a7da2ac99e143d0b47e137026874ad367938c89b04198bee42  dogecoin-1.14.7-win32-debug.zip
b5fbc247f1742550c1199864070d9e2be525e9441845f4e6ad23f47b8de23ab3  dogecoin-1.14.7-win32-setup-unsigned.exe
a7a9ab7784a264943a34d2fba9ee054c49ec3cd68c328fc5ad9b7653133f78bf  dogecoin-1.14.7-win32.zip
27d65e3da84a75ef8b98155d96d833b453f5b87db87db8adf535d54765ba96fc  dogecoin-1.14.7-win64-debug.zip
dd5d911f3e651737c0b3d8bd5d06aa96ed7c87b1411407721e4cc9662869e764  dogecoin-1.14.7-win64-setup-unsigned.exe
f282c8ad91bf08e15e9fcdde78e688af68314a3e4abd5aacd2e04545bc66d2f2  dogecoin-1.14.7-win64.zip
5c03f091fbd8e47d0e4f485a8c6bb94b6f20b3c0580fc36e0dda46b0412f5187  dogecoin-1.14.7-aarch64-linux-gnu-debug.tar.gz
200ae964e565afc9e3b5715d400629c9f7bc14f539b62849fb11444da4bf2849  dogecoin-1.14.7-aarch64-linux-gnu.tar.gz
b093fb7c1cef6d6c4df3e7e93e4b43ed347a9543c84b078550aea18234ba2d52  dogecoin-1.14.7-arm-linux-gnueabihf-debug.tar.gz
640fed354b3a13a4cc0ac00d6d5c4d9150fe9bc29ac808a3347d99933675c14c  dogecoin-1.14.7-arm-linux-gnueabihf.tar.gz
45804db60cd9472af16d78f7cbd98fe82bb32039f5b6232ae881e7c79da03a2a  dogecoin-1.14.7-i686-pc-linux-gnu-debug.tar.gz
881fa1086324350127ba618e86ca0bc3977ca8d8bd5fc2b2def46b6a63533bc4  dogecoin-1.14.7-i686-pc-linux-gnu.tar.gz
4d4e04e8f0cddb240c5927e758d8c17aae0859f3a171b2ec9f538756fd960edc  dogecoin-1.14.7-x86_64-linux-gnu-debug.tar.gz
182e0240121bf50a28ed75ff956fbd6364edb93823d2514a3b77a5561aa32624  dogecoin-1.14.7-x86_64-linux-gnu.tar.gz
```

**Comment by xanimo** [comment]
ACK, expected results on 64-bit win/nix for both focal and bionic:
https://gist.github.com/xanimo/00de9c628262cdadd2919378fb71c2b4

File: contrib/devtools/symbol-check.py
Comment (chromatic) [inline contrib/devtools/symbol-check.py:117]: This library appears to manage color matching with things like ICC profiles. Today I learned!

**Comment by chromatic** [comment]
Fine with me to squash as appropriate.

I'm curious but don't want to hold up this commit; I see Python 3 type annotations. Are we using `mypy` to validate these anywhere? That's probably a big can of worms, but I can see its value in the future.

File: contrib/devtools/symbol-check.py
Comment (chromatic) [reply contrib/devtools/symbol-check.py:117]: A Qt 5.15.x will take some work. [Bitcoin's bump to Qt 5.15.5](https://github.com/bitcoin/bitcoin/commit/ecb617fefe9fbd15365dcf0469d4d39df6b856bc) has a few patches we should review. My instincts tell me that would delay a 1.14.7 release, so I recommend:

 * start the Qt upgrade early in the next minor release cycle
 * add it to the "let's port GUI changes" stage for the 1.21 series

**Comment by chromatic** [comment]
Before I kick off a gitian run, I'm trying this locally on 64-bit x86 Linux (Ubuntu 22.04). I have repeatable RPC test failures, all around timeouts:

```
$ python3 qa/rpc-tests/sendheaders.py 
Initializing test directory /tmp/testfmlomx3g/969772
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:13162
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:13162
Part 1: headers don't start before sendheaders message...
Assertion failed: 
  File "/home/chromatic/dev/dogecoin/qa/rpc-tests/test_framework/test_framework.py", line 145, in main
    self.run_test()
  File "/home/chromatic/dev/dogecoin/qa/rpc-tests/sendheaders.py", line 312, in run_test
    test_node.wait_for_getdata([new_block.sha256], timeout=25)
  File "/home/chromatic/dev/dogecoin/qa/rpc-tests/sendheaders.py", line 191, in wait_for_getdata
    assert(wait_until(test_function, timeout=timeout))
Stopping nodes
Not cleaning up dir /tmp/testfmlomx3g/969772
Failed
```

I'm not sure what in the commits would cause this.

**Comment by chromatic** [comment]
Good news; I was checking versions to get back to you and found that I had a different `python3` on my path than what I expected. Switching versions gave me a clean test run. I'll get back to gitian soon.

**Comment by chromatic** [comment]
It was a Python 3.9 statically linked against different OpenSSL libraries. Long story. I'm not worried about Python 3.11 yet.

**Comment by chromatic** [comment]
Code ACK on Jammy, 64-bit x86. I'll make a Gitian build next.

**Comment by chromatic** [comment]
Here's what I have from gitian so far.

```
7a2390dda315656ddf6c0a2562ebf48e913db168c0498225fa4ed39a367a9506  dogecoin-1.14.7-win-unsigned.tar.gz
3eb24ccbd3b8b7a7da2ac99e143d0b47e137026874ad367938c89b04198bee42  dogecoin-1.14.7-win32-debug.zip
3c87af83e228e7932ad4ba22ab9c9105c2d91cdee60ed0ba6a9d76f8f5be284a  dogecoin-1.14.7-win32-setup-unsigned.exe
b8c15106233e4de6d8ff72cc28203e63d630e39a7c00d23d074f600bc29135c3  dogecoin-1.14.7-win32.zip
27d65e3da84a75ef8b98155d96d833b453f5b87db87db8adf535d54765ba96fc  dogecoin-1.14.7-win64-debug.zip
0bc3a92a9489fb5682b38b9032e91b5a17e23f24e86089e6bc8e64d2b591ff80  dogecoin-1.14.7-win64-setup-unsigned.exe
d4fadf84433e055a7993378bb9797aa31a699a32ae81eea30ec6d4ccb3fabfe5  dogecoin-1.14.7-win64.zip
8e5064748145a08b5ca1257af31c738b4058b7549dad4d372f67a855d1edb1d4  src/dogecoin-1.14.7.tar.gz
17df0ff9c6fcc2dcba4f8bd5f9398293bf575d2f0a5e839b93ab2be1f601fee3  dogecoin-win-1.14-res.yml
7a081770c6cf4e00d913b09fd4de253ee4657ec1b7a3bff49fd170bba3325a0b  dogecoin-1.14.7-osx-unsigned.dmg
4055928d2dd21a32b0866728f05e1fdda85929cc28bf8d04bc961faf24f95471  dogecoin-1.14.7-osx-unsigned.tar.gz
50fd23672cb2e1558fa490e29d84151431bc84b35b29bf0593eb42f496cffede  dogecoin-1.14.7-osx64.tar.gz
8e5064748145a08b5ca1257af31c738b4058b7549dad4d372f67a855d1edb1d4  src/dogecoin-1.14.7.tar.gz
bd9ce437f1cb5a88e5f82d46dad14f01ee2f4a634d4764e6870bfca2e618f4a6  dogecoin-osx-1.14-res.yml
```

**Comment by xanimo** [comment]
Here are mine from docker:
```
bf1bb2cc23796fd0a2aa629a34432dbcbdb8cfe0e86ad9375c3dc464a4ce1841  dogecoin-1.14.7-aarch64-linux-gnu-debug.tar.gz
8a4b16612e9ca37d0acd5ead4291a2f42918a57ac2063a81ec7fe70f402c74a2  dogecoin-1.14.7-aarch64-linux-gnu.tar.gz
0839350d4e89699f1a75da767f3a49617d1a05fc3c9bb31018b64822170c4d7f  dogecoin-1.14.7-arm-linux-gnueabihf-debug.tar.gz
ad9e883608d7d7e313f9b0471b4ca8af031452f0fa4067700c4e04711b258d41  dogecoin-1.14.7-arm-linux-gnueabihf.tar.gz
833dc207426a90828f0748736a2d9a332db9a3910f0c858e9f1adf765443100f  dogecoin-1.14.7-i686-pc-linux-gnu-debug.tar.gz
8b2f57a3351480ca0a0c5bb98f6ef08f286ac6e3272afaaa876287f08dbff98f  dogecoin-1.14.7-i686-pc-linux-gnu.tar.gz
b55e3582def8c89a660b321e3f8b6dec9653cd2c0545217fc46be5fba9547b62  dogecoin-1.14.7-x86_64-linux-gnu-debug.tar.gz
1d5ecb07ea7a2352fe19962f9005c8f743f836696b5b8f8e3765141d31ab0e46  dogecoin-1.14.7-x86_64-linux-gnu.tar.gz
7a2390dda315656ddf6c0a2562ebf48e913db168c0498225fa4ed39a367a9506  dogecoin-1.14.7-win-unsigned.tar.gz
3eb24ccbd3b8b7a7da2ac99e143d0b47e137026874ad367938c89b04198bee42  dogecoin-1.14.7-win32-debug.zip
3c87af83e228e7932ad4ba22ab9c9105c2d91cdee60ed0ba6a9d76f8f5be284a  dogecoin-1.14.7-win32-setup-unsigned.exe
b8c15106233e4de6d8ff72cc28203e63d630e39a7c00d23d074f600bc29135c3  dogecoin-1.14.7-win32.zip
27d65e3da84a75ef8b98155d96d833b453f5b87db87db8adf535d54765ba96fc  dogecoin-1.14.7-win64-debug.zip
0bc3a92a9489fb5682b38b9032e91b5a17e23f24e86089e6bc8e64d2b591ff80  dogecoin-1.14.7-win64-setup-unsigned.exe
d4fadf84433e055a7993378bb9797aa31a699a32ae81eea30ec6d4ccb3fabfe5  dogecoin-1.14.7-win64.zip
7a081770c6cf4e00d913b09fd4de253ee4657ec1b7a3bff49fd170bba3325a0b  dogecoin-1.14.7-osx-unsigned.dmg
4055928d2dd21a32b0866728f05e1fdda85929cc28bf8d04bc961faf24f95471  dogecoin-1.14.7-osx-unsigned.tar.gz
50fd23672cb2e1558fa490e29d84151431bc84b35b29bf0593eb42f496cffede  dogecoin-1.14.7-osx64.tar.gz
8e5064748145a08b5ca1257af31c738b4058b7549dad4d372f67a855d1edb1d4  src/dogecoin-1.14.7.tar.gz
```

**Comment by chromatic** [comment]
I'm looking at this again now. If I can get a Linux build, I'll match those checksums. Mac and Windows match, so this should be the last piece.

**Comment by chromatic** [comment]
Linux gitian checksums. I think we're good to go.

```
bf1bb2cc23796fd0a2aa629a34432dbcbdb8cfe0e86ad9375c3dc464a4ce1841  dogecoin-1.14.7-aarch64-linux-gnu-debug.tar.gz
8a4b16612e9ca37d0acd5ead4291a2f42918a57ac2063a81ec7fe70f402c74a2  dogecoin-1.14.7-aarch64-linux-gnu.tar.gz
0839350d4e89699f1a75da767f3a49617d1a05fc3c9bb31018b64822170c4d7f  dogecoin-1.14.7-arm-linux-gnueabihf-debug.tar.gz
ad9e883608d7d7e313f9b0471b4ca8af031452f0fa4067700c4e04711b258d41  dogecoin-1.14.7-arm-linux-gnueabihf.tar.gz
833dc207426a90828f0748736a2d9a332db9a3910f0c858e9f1adf765443100f  dogecoin-1.14.7-i686-pc-linux-gnu-debug.tar.gz
8b2f57a3351480ca0a0c5bb98f6ef08f286ac6e3272afaaa876287f08dbff98f  dogecoin-1.14.7-i686-pc-linux-gnu.tar.gz
b55e3582def8c89a660b321e3f8b6dec9653cd2c0545217fc46be5fba9547b62  dogecoin-1.14.7-x86_64-linux-gnu-debug.tar.gz
1d5ecb07ea7a2352fe19962f9005c8f743f836696b5b8f8e3765141d31ab0e46  dogecoin-1.14.7-x86_64-linux-gnu.tar.gz
8e5064748145a08b5ca1257af31c738b4058b7549dad4d372f67a855d1edb1d4  src/dogecoin-1.14.7.tar.gz
0245f71626b2f99ccc90dcc959c95ae7661ed390e60a3e839a7c6696f3b3668c  dogecoin-linux-1.14-res.yml
```

**Review by chromatic — APPROVED** [review]
With the code review comments addressed and gitian check passing and signatures verified by three different builds, I'm satisfied. ACK.

**Pull request merged**
